### PR TITLE
document deprecation of `@uppy/aws-s3-multipart`

### PR DESCRIPTION
**Pull request by aduh95**

### CODE REVIEW

File: docs/uploader/aws-s3-multipart.mdx [hunk 1]
```diff
@@ -163,20 +169,49 @@ Drive). Otherwise it’s not worth the hosting effort.
...
 ### Options
 
+#### `shouldUseMultipart(file)`
+
+:::warning
+
+You must define this option, otherwise you’ll be using the
+[legacy version of this plugin](./aws-s3). If you use the deprecated
+`@uppy/aws-s3-multipart` package, this will be set to `true` by default, which
+is not a recomended setting.
```
Comment (Murderlon) [inline docs/uploader/aws-s3-multipart.mdx:193]:
```suggestion
Not setting this option will result in a suboptimal experience for some of your user’s uploads.
It is best for speed and stability to upload large (100&nbsp;MB+) files with multipart
and small files with regular uploads.
```

File: docs/uploader/aws-s3-multipart.mdx [hunk 2]
```diff
@@ -163,20 +169,49 @@ Drive). Otherwise it’s not worth the hosting effort.
...
+
+:::
+
+A boolean, or a function that returns a boolean which is called for each file
+that is uploaded with the corresponding `UppyFile` instance as argument.
+
+By default, all files are uploaded as multipart. In a future version, all files
+with a `file.size` ≤ 100 MiB will be uploaded in a single chunk, all files
```
Comment (Murderlon) [inline docs/uploader/aws-s3-multipart.mdx:201]:
```suggestion
with a `file.size` ≤ 100&nbsp;MB will be uploaded in a single chunk, all files
```

File: docs/uploader/aws-s3-multipart.mdx [hunk 3]
```diff
@@ -163,20 +169,49 @@ Drive). Otherwise it’s not worth the hosting effort.
...
+larger than that as multipart.
+
+Here’s how to use it:
+
+```js
+uppy.use(AwsS3, {
+	shouldUseMultipart(file) {
+		// Use multipart only for files larger than 100MiB.
```
Comment (Murderlon) [inline docs/uploader/aws-s3-multipart.mdx:209]:
```suggestion
		// Use multipart only for files larger than 100MB
```

File: docs/uploader/aws-s3-multipart.mdx [hunk 4]
```diff
@@ -7,10 +7,12 @@ import Tabs from '@theme/Tabs';
...
 
-# AWS S3 Multipart
+# AWS S3
 
-The `@uppy/aws-s3-multipart` plugin can be used to upload files directly to an
-S3 bucket (or S3-compatible provider) using S3’s Multipart uploader.
+The `@uppy/aws-s3` plugin can be used to upload files directly to a S3 bucket or
+a S3-compatible provider, such as Google Cloud Storage or DigitalOcean Spaces.
+Uploads can be signed using either [Companion][companion docs] or a custom
+signing function.
```
[the remark this answers is not in the document]
Comment (aduh95) [reply docs/uploader/aws-s3-multipart.mdx:15]: Maybe we should put the custom function at the end.
```suggestion
Uploads can be signed using either [Companion][companion docs], temporary credentials, or a custom
signing function.
```

File: docs/uploader/aws-s3-multipart.mdx (hunk 1 above)
Comment (aduh95) [reply docs/uploader/aws-s3-multipart.mdx:193]: We probably want to keep the link to the legacy docs, no?

File: docs/uploader/aws-s3-multipart.mdx (hunk 2 above)
Comment (aduh95) [reply docs/uploader/aws-s3-multipart.mdx:201]: We'll be using 100MiB, not 100 MB.
```suggestion
with a `file.size` ≤ 100&nbsp;MiB will be uploaded in a single chunk, all files
```

File: docs/uploader/aws-s3-multipart.mdx (hunk 3 above)
Comment (aduh95) [reply docs/uploader/aws-s3-multipart.mdx:209]: `100 * 2 ** 20` is 100 MiB, not 100 MB

File: docs/uploader/aws-s3-multipart.mdx (hunk 1 above)
Comment (Murderlon) [reply docs/uploader/aws-s3-multipart.mdx:193]:
```suggestion
Until the next major version, not setting this option uses the [legacy version of this plugin](./aws-s3).
This is a suboptimal experience for some of your user’s uploads.
It is best for speed and stability to upload large (100&nbsp;MB+) files with multipart
and small files with regular uploads.
```

**Review by Murderlon — APPROVED** [review]

**Comment by aduh95** [comment]
@arturi can you review the policy documented here? I wonder if it's still up-to-date with client signing requirements.

File: docs/guides/choosing-uploader.md [hunk 5]
```diff
@@ -56,24 +56,8 @@ cases be preferable, for instance, to reduce costs or the complexity of running
...
-network issues (only the failed parts need to be retried). The downside is
-request overhead, as it needs to do creation, signing, and completion requests
-besides the upload requests. For example, if you are uploading files that are
-only a couple kilobytes with a 100ms roundtrip latency, you are spending 400ms
-on overhead and only a few milliseconds on uploading.
-
-If you are uploading large files (100&nbsp;MB+), we recommend
-[`@uppy/aws-s3-multipart`][], otherwise [`@uppy/aws-s3`][].
+[`@uppy/aws-s3-multipart`][], but we are planning to merge the two plugins in
+the next major. You should use [`@uppy/aws-s3`][] with the new `shouldUseMultipart` option.
```
Comment (aduh95) [inline docs/guides/choosing-uploader.md:60]:
```suggestion
the next major. You should use [`@uppy/aws-s3`][] with the new
`shouldUseMultipart` option.
```

**Pull request merged**